model_tools/brain_transformation/behavior.py

```diff
@@ -215,3 +216,54 @@ def labels_to_indices(self, labels):
                 indices.append(label2index[label])
             index2label = OrderedDict((index, label) for label, index in label2index.items())
             return indices, index2label
+        
+
+class OddOneOutBehavior(BrainModel):
+    def __init__(self, identifier: str, activations_model, layer: Union[str, List[str]], similarity_measure: str = 'dot'):
+        """
+        :param identifier: a string to identify the model
+        :param activations_model: the model from which to retrieve representations for stimuli
+        :param layer: the single behavioral readout layer or a list of layers to read out of.
+        """
+        self._identifier = identifier
+        self.activations_model = activations_model
+        self.readout = make_list(layer)
+        self.current_task = None
+        self.simmilarity_measure = similarity_measure
+
+    @property
+    def identifier(self):
+        return self._identifier
+
+    def start_task(self, task: BrainModel.Task):
+        assert task == BrainModel.Task.odd_one_out
+        self.current_task = task
+
+    def look_at(self, stimuli, number_of_trials=1):
+        assert self.current_task == BrainModel.Task.odd_one_out
+        features = self.activations_model(stimuli, layers=self.readout)
+        features = features.transpose('presentation', 'neuroid')
+        odd_one_out = odd_one_out(features)
+        odd_one_out = BehavioralAssembly(odd_one_out, coords={None}, dims=[None])  # TODO
+        return odd_one_out
+    
+    def odd_one_out(self, features):
+        if self.similarity_measure == 'dot':
+            similarity_01 = np.dot(features[0], features[1])
+            similarity_02 = np.dot(features[0], features[2])
+            similarity_12 = np.dot(features[1], features[2])
```

Comment:
```suggestion
        stimuli = features['stimulus_id'].values
        if self.similarity_measure == 'dot':
            similarity_values = np.einsum('ij,ij->i', features, np.roll(features, 1, axis=0))
            similarities = DataAssembly(similarity_values, coords={'stimulus_left': ('presentation', stimuli), 'stimulus_right': ('presentation', np.roll(stimuli, 1))}, dims=['presentation'])
```
This will give something like
```
<xarray.DataAssembly(presentation: 3)>
array([ 8,  5, 10])
Coordinates:
    stimulus_left   (presentation) <U5 'stim1' 'stim2' 'stim3'
    stimulus_right  (presentation) <U5 'stim3' 'stim1' 'stim2'
```
such that you don't have to worry about positional indexing.